src/Analyzers/CSharp/Analysis/InvocationExpressionAnalyzer.cs

```diff
@@ -192,6 +192,23 @@ private static void AnalyzeInvocationExpression(SyntaxNodeAnalysisContext contex
                                 break;
                             }
                         case "Last":
+                            {
+                                if (DiagnosticRules.OptimizeLinqMethodCall.IsEffective(context))
+                                {
+                                    if (!invocationInfo.Expression.IsKind(SyntaxKind.ElementAccessExpression, SyntaxKind.InvocationExpression)
+                                        && UseElementAccessAnalysis.IsFixableLast(invocationInfo, context.SemanticModel, context.CancellationToken))
```

Comment:
Method `IsFixableLast` should contain a check if current compilation supports C# 8.0 or higher